core/modules/filter/filter.pages.inc

```diff
@@ -410,6 +410,7 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
     '#parents' => array('fid'),
     '#weight' => -2,
     '#access' => $element_supports_uploads && $upload_settings['status'] && user_access('upload editor files'),
+    '#description' => t('Files must be less than @size. <br /> Allowed file types: @extensions.', array('@size' => format_size($max_filesize), '@extensions' => $file_extensions)),
```

Comment:
This should call `theme('theme_file_upload_help')` to print this message. I'll do a reroll.